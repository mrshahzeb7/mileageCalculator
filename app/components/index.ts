export * from './StatusBar/StatusBar';
export * from './BottomTab/BottomTab';
export * from './ScreenHeaders';
export * from './Icon/Icon';
export * from './AddButton/AddButton';
export * from './HomeList/HomeListItem';
export * from './HomeList/HomeListSectionHeader';
export * from './TextField/TextField';
export * from './Label/Label';
export * from './Picker/Picker';
export * from './DatePicker/DatePicker';
export * from './RowInput/RowInput';
export * from './ComponentTree/ComponentTree';
export * from './TimeLineList/TimeLineList';
export * from './Dot/Dot';
export * from './Loader/Loader';
