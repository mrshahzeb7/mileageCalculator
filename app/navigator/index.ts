export * from './routeNames';
export * from './AppNavigator/AppNavigator';
