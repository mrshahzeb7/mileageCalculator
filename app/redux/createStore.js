import {applyMiddleware, compose, createStore} from 'redux';

export default (rootReducer, rootSaga) => {
  const middleware = [];
  const enhancers = [];

  enhancers.push(applyMiddleware(...middleware));

  const createAppropriateStore = createStore;

  const store = createAppropriateStore(rootReducer, compose(...enhancers));

  return {
    store,
  };
};
