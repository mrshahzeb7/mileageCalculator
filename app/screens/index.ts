export * from './AddEntry/AddEntry';
export * from './Home/Home';
export * from './TimeLine/TimeLine';
