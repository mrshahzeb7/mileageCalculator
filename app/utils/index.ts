export * from './responsiveDimension/responsiveDimension';
export * from './utils';
export * from './strings';
