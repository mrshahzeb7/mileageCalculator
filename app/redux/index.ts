import {combineReducers} from 'redux';
import configureStore from './createStore';
import MainActions, {reducer as mainReducer} from './mainRedux/mainRedux';
import NavigationActions, {
  reducer as navigationReducer,
} from './navigationRedux/navigationRedux';

/* ------------- Assemble The Reducers ------------- */
export const reducers = combineReducers({
  navigation: navigationReducer,
  main: mainReducer,
});

import {persistReducer, persistStore} from 'redux-persist';
import storage from '@react-native-async-storage/async-storage'; // defaults to localStorage for web

const persistConfig = {
  key: 'root',
  storage,
};

const persistedReducer = persistReducer(persistConfig, reducers);

export default () => {
  let {store} = configureStore(persistedReducer);
  let persistor = persistStore(store);
  return {store, persistor};
};

export * from './hooks';
export {MainActions, NavigationActions};
