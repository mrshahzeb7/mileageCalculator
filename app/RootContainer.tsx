import React from 'react';
import {StyleSheet, View} from 'react-native';
import {BottomTab, MainHeader, StatusBar} from './components';

import {AppNavigator} from './navigator';
import {AddEntry} from './screens';
import {colors} from './theme';

interface Props {}

export const RootContainer: React.FC<Props> = () => {
  return (
    <View style={styles.container}>
      <MainHeader />
      <StatusBar />
      <AppNavigator />
      <AddEntry />
      <BottomTab />
    </View>
  );
};

// styles
const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'space-between',
    backgroundColor: colors.background,
  },
});
