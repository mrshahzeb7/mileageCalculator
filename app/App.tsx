import React from 'react';
import {StyleSheet, View} from 'react-native';
import {
  SafeAreaProvider,
  initialWindowMetrics,
} from 'react-native-safe-area-context';
import {RootContainer} from './RootContainer';
import {colors} from './theme';
import {Provider} from 'react-redux';
import createStore from './redux';
import {PersistGate} from 'redux-persist/integration/react';

const {store, persistor} = createStore();
interface Props {}
export const App: React.FC<Props> = () => {
  return (
    <View style={styles.container}>
      <SafeAreaProvider initialMetrics={initialWindowMetrics}>
        <Provider store={store}>
          <PersistGate loading={null} persistor={persistor}>
            <RootContainer />
          </PersistGate>
        </Provider>
      </SafeAreaProvider>
    </View>
  );
};

// styles
const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'space-between',
    backgroundColor: colors.background,
  },
});
