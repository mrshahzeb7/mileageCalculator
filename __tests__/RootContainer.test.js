/**
 * @format
 */

import 'react-native';
import React from 'react';
import {RootContainer} from '../app/RootContainer';

// Note: import explicitly to use the types shiped with jest.
import {it} from '@jest/globals';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import {
  SafeAreaProvider,
  initialWindowMetrics,
} from 'react-native-safe-area-context';
import {Provider} from 'react-redux';
import createStore from '../app/redux';
const store = createStore();

const Parent = () => {
  return (
    <Provider store={store}>
      <SafeAreaProvider initialMetrics={initialWindowMetrics}>
        <RootContainer />
      </SafeAreaProvider>
    </Provider>
  );
};

it('app renders correctly', () => {
  const tree = renderer.create(<Parent />).toJSON();
  expect(tree).toMatchSnapshot();
});
