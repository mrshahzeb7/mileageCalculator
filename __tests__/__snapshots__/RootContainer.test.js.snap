// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app renders correctly 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    {
      "flex": 1,
    }
  }
/>
`;
